pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table.go

```diff
@@ -11,12 +11,42 @@
 package scbuildstmt
 
 import (
+	"reflect"
+
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
 	"github.com/cockroachdb/cockroach/pkg/sql/privilege"
 	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scerrors"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/errors"
 )
 
+// supportedAlterStatements tracks alter table operations fully supported by
+// declarative schema  changer. Operations marked as non-fully supported can
+// only be with the experimental_use_new_schema_changer session variable.
+var supportedAlterStatements = map[reflect.Type]supportedStatement{
+	reflect.TypeOf((*tree.AlterTableAddColumn)(nil)): {alterTableAddColumn, false},
+}
```

Comment:
Nice! I like this pattern. That being said, you might want to rename `supportedAlterStatements` into `supportedAlterTableStatements` instead, because `ALTER TYPE` or `ALTER DATABASE` commands won't end up in here.